### ISSUE TITLE
Wrong date applied to backup file name when created during midnight hour (12-1 a.m.)

### ISSUE THREAD

**Issue by gerryfrancis** [deltachat/deltachat-android#2126]
- Android version:
Android 11.

- Device:
Fairphone FP2 (Lineage OS 18.1 build RQ3A.210705.001 from 2021-11-05).

- Delta Chat version:
1.23.1 (nightly build from 2021-11-05).

- Expected behavior:
The backup file name includes the date of the day the backup was created.

- Actual behavior:
The backup file name includes the date of yesterday before the backup was created.

- Steps to reproduce the problem:
-- Wait until midnight (12 a.m.).
-- Create a backup ("Settings" -> "Chats and Media" -> "Back Up Chats to External Storage" -> "Start Backup"). (Result: The file name includes the date of yesterday.)

- Screenshots:
N/A.

- Logs:
N/A.

- Remarks:
-- When a log file is saved during 12-1 a.m., the date applied to the file name is correct.
-- Android daylight savings settings obviously have no impact.

**Comment by r10s** [comment]
the timestamp in the filename is "UTC date and time", in germany/austria, the local time is one hour ahead.

the corresponding code line looks good to me , https://github.com/deltachat/deltachat-core-rust/blob/50a5e715d2652fde684e610243604ada862a61a8/src/imex.rs#L671 , the used `time()` call is used throughout the app, making these timestamps comparable, i think, this is a huge advantage. also when travelling or getting backups from other timezones or whatnot. not to mention that backup naming would really fail when it comes to daylight sacving (in case there is the same local time twice...). using local time has lots of implications :)

so, unless i have overseen something, i think, things are okay.

**Comment by gerryfrancis** [comment]
> the timestamp in the filename is "UTC date and time"

@r10s No, the format is "delta-chat-backup-YYYY-MM-DD-xx.tar", so no time included.

> in germany/austria, the local time is one hour ahead

@r10s According to https://de.wikipedia.org/wiki/Koordinierte_Weltzeit there are two hours difference, so UTC is still 2021-11-07 23:51 as I write this comment.

> the used time() call is used throughout the app, making these timestamps comparable

@r10s As the file names do not include a timestamp, confusion may occur to the user because an earlier date indicated in the file name does not comply with the timestamp of the file itself. Although Austria/Germany have only two hours difference to UTC, confusion might get worse for Delta Chat users living in areas with time zones of higher difference to UTC.

@r10s Following your argumentation, the log file would be affected from this issue, too, but it is not. It might be understandable from a developers point of view, but not for a common user.

**Comment by r10s** [comment]
you are right, there is not time in the filename, only date. wondering if there may be issues if we switch away from UTC in that case. or if it isn't better to stay with UTC.

(also moving that to deltachat-core-rust as there is nothing android can do about that)
